ARCHITECTURE.md

```diff
@@ -0,0 +1,71 @@
+# Architecture
+
+Rojo is a rather large project with a bunch of moving parts. While it's not too complicated in practice, it tends to be overwhelming because it's a fair bit of Rust and not very clear where to begin.
+
+This document is a "what the heck is going on" level view of Rojo and the codebase that's written to make it more reasonable to jump into something. It won't go too into depth on *how* something is done, but it will go into depth on *what* is being done.
+
+## Overarching
+
+Rojo is divided into two main pieces: the server and the plugin. The server is what's ran on your computer (whether it be via the terminal or the visual studio code extension), with the plugin serving as its main client.
```

Comment:
```suggestion
Rojo is divided into two main pieces: the server and the plugin. The server is what's ran on your computer (whether it be via the terminal, the visual studio code extension, or a remote machine), with the plugin serving as its main client.
```
Perhaps worth mentioning that it can work over a remote connection too?